include/capstone/riscv.h

```diff
@@ -0,0 +1,498 @@
+#ifndef CAPSTONE_RISCV_H
+#define CAPSTONE_RISCV_H
+
+/* Capstone Disassembly Engine */
+/* RISC-V Backend By Rodrigo Cortes Porto <porto703@gmail.com> & 
+   Shawn Chang <citypw@gmail.com>, HardenedLinux@2018 */
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#if !defined(_MSC_VER) || !defined(_KERNEL_MODE)
+#include <stdint.h>
+#endif
+
+#include "capstone/platform.h"
+
+// GCC MIPS toolchain has a default macro called "mips" which breaks
+// compilation
+//#undef riscv
+
+#ifdef _MSC_VER
+#pragma warning(disable:4201)
+#endif
+
+//> Operand type for instruction's operands
+typedef enum riscv_op_type {
+	RISCV_OP_INVALID = 0, // = CS_OP_INVALID (Uninitialized).
+	RISCV_OP_REG, // = CS_OP_REG (Register operand).
+	RISCV_OP_IMM, // = CS_OP_IMM (Immediate operand).
+	RISCV_OP_MEM, // = CS_OP_MEM (Memory operand).
+} riscv_op_type;
+
+// Instruction's operand referring to memory
+// This is associated with RISCV_OP_MEM operand type above
+typedef struct riscv_op_mem {
+	unsigned int base;	// base register
+	int64_t disp;	// displacement/offset value
+} riscv_op_mem;
+
+// Instruction operand
+typedef struct cs_riscv_op {
+	riscv_op_type type;	// operand type
+	union {
+		unsigned int reg;	// register value for REG operand
+		int64_t imm;		// immediate value for IMM operand
+		riscv_op_mem mem;	// base/disp value for MEM operand
+	};
+} cs_riscv_op;
+
+// Instruction structure
+typedef struct cs_riscv {
+	// Number of operands of this instruction, 
+	// or 0 when instruction has no operand.
+	uint8_t op_count;
+	cs_riscv_op operands[8]; // operands for this instruction.
```

Comment:
can RISCV have up to 8 operands? if not, then this should be updated.

Reply:
maybe, It depends on the riscv _Packed-SIMD Instructions_ or some other extensions, for now it is only version 0.1.